spec-dtslint/operators/filter-spec.ts

```diff
@@ -56,4 +56,17 @@ it('should support inference from a return type with Boolean as a predicate', ()
 
   const i$: Observable<I> = of();
   const s$: Observable<string> = i$.pipe(map(i => i.a), filter(Boolean)); // $ExpectType Observable<string>
-});
\ No newline at end of file
+});
+
+it('should support inference from a generic return type of the predicate', () => {
+  function isDefined<T>() {
+    return (value: T|undefined|null): value is T => {
+      return value !== undefined && value !== null;
+    };
+  }
+
+  // $ExpectType Observable<string>
+  const source = of('World', 'test', null, 'ok').pipe(
```

Comment:
Does this work with more than one type?

```
const o$ = of(1, true, 'Nick Cage').pipe(filter(isDefined())) // $ExpectType Observable<number | boolean | string>
```

Reply:
Yup, updated the test to the following to reflect that:

```ts
  const o$ = of(1, null, {foo: 'bar'}, true, undefined, 'Nick Cage').pipe(filter(isDefined())); // $ExpectType Observable<string | number | boolean | { foo: string; }>

```